gammapy/background/make.py

```diff
@@ -0,0 +1,470 @@
+        Cube background model to smooth.
+    n_counts : int
+        Number of events used to fill the cube background model.
+
+    Returns
+    -------
+    bg_cube : `~gammapy.background.CubeBackgroundModel`
+        Smoothed cube background model.
+    """
+    from scipy import ndimage
+
+    # smooth images
+
+    # integral of original images
+    dummy_delta_energy = np.zeros_like(bg_cube.energy_bins[:-1])
+    delta_y = bg_cube.dety_bins[1:] - bg_cube.dety_bins[:-1]
+    delta_x = bg_cube.detx_bins[1:] - bg_cube.detx_bins[:-1]
+    # define grid of deltas (i.e. bin widths for each 3D bin)
+    dummy_delta_energy, delta_y, delta_x = np.meshgrid(dummy_delta_energy, delta_y,
+                                                       delta_x, indexing='ij')
+    bin_area = (delta_y*delta_x).to('sr')
+    integral_image = bg_cube.background*bin_area
+    integral_image = integral_image.sum(axis=(1,2))
+
+    # number of times to smooth
+    if n_counts >= 1.e6:
+        n_smooth = 3
+    elif (n_counts < 1.e6) and (n_counts >= 1.e5):
+        n_smooth = 4
+    else:
+        n_smooth = 5
+
+    # smooth images
+
+    # define smoothing kernel as k5a in root:
+    # https://root.cern.ch/root/html/TH2.html#TH2:Smooth
+    kernel = np.array([[0, 0, 1, 0, 0],
+                       [0, 2, 2, 2, 0],
+                       [1, 2, 5, 2, 1],
+                       [0, 2, 2, 2, 0],
+                       [0, 0, 1, 0, 0]])
+
+    # loop over energy bins (i.e. images)
+    for i_energy in np.arange(len(bg_cube.energy_bins) - 1):
+        # loop over number of times to smooth
+        for i_smooth in np.arange(n_smooth):
+            data = bg_cube.background[i_energy]
+            image_smooth = ndimage.convolve(data, kernel)
+
+            # overwrite bg image with smoothed bg image
+            bg_cube.background[i_energy] = Quantity(image_smooth, bg_cube.background.unit)
+
+    # integral of smooth images
+    integral_image_smooth = bg_cube.background*bin_area
+    integral_image_smooth = integral_image_smooth.sum(axis=(1,2))
+
+    # scale images to preserve original integrals
+
+    # loop over energy bins (i.e. images)
+    for i_energy in np.arange(len(bg_cube.energy_bins) - 1):
+        bg_cube.background[i_energy] *= (integral_image/integral_image_smooth)[i_energy]
+
+    return bg_cube
+
+
+def make_bg_cube_model(observation_table, fits_path, DEBUG):
+    """Create a bg model from an observation table.
+
+    Produce a background cube using the data from an observation list.
+    Steps:
+
+    1. define binning
+    2. fill events and livetime correction in cubes
+    3. fill events in bg cube
+    4. smooth
+    5. correct for livetime and bin volume
+    6. set 0 level
+
+    TODO: review steps!!!
+
+    Parameters
+    ----------
+    observation_table : `~gammapy.obs.ObservationTable`
+        Observation list to use for the histogramming.
+    fits_path : str
+        Path to the data files.
+    DEBUG : int
```

Comment:
Again ... it's important to use Python logging.

This function could be one of those cases I'm talking about in the last paragraph [here](https://gammapy.readthedocs.org/en/latest/development/index.html#generating-log-messages) where it makes sense that they optionally take their own logger as input. [Example](https://gammapy.readthedocs.org/en/latest/api/gammapy.data.EventListDatasetChecker.html).